tornado/ioloop.py

```diff
@@ -898,11 +899,13 @@ def is_running(self) -> bool:
         """
         return self._running
 
-    def _run(self) -> None:
+    async def _run(self) -> None:
         if not self._running:
             return
         try:
-            return self.callback()
+            val = self.callback()
+            if isawaitable(val):
+                await val  # type: ignore
```

Comment:
The reason you needed a `type: ignore` here is because `PeriodicCallback.__init__` specifies that the callback is a function returning None. You need to update that annotation to allow coroutines as well, or else every user of this functionality will need `type: ignore` (if they're using mypy).